pkgs/stdenv/generic/default.nix

```diff
@@ -48,6 +48,10 @@ let lib = import ../../../lib; in lib.makeOverridable (
 
 , # The platform which build tools (especially compilers) build for in this stage,
   targetPlatform
+
+, # The implementation of `mkDerivation`, parameterized with the final stdenv so we can tie the knot.
+  # This is convient to have as a parameter so the stdenv "adapters" work better
+  mkDerivationFromStdenv ? import ./make-derivation.nix { inherit lib config; }
```

Comment:
Is it necessary to give a default value given that `mkDerivationFromStdenv-super` defaults to `defaultMkDerivationFromStdenv` in `adapters.nix`?

Reply:
The non-adapted stdenv needs this. I sort of see it as the normal one and the other as the unfortunate cludge too.